dlp-terraform/ecs/ecs_training_service.tf

```diff
@@ -0,0 +1,212 @@
+resource "aws_ecs_task_definition" "training" {
+  family             = "training"
+  task_role_arn      = aws_iam_role.ecs_task_role.arn
+  execution_role_arn = aws_iam_role.ecs_exec_role.arn
+  network_mode       = "bridge"
+  cpu                = 1024
+  memory             = 4096
+
+  container_definitions = jsonencode([
+    {
+      "name" : "training",
+      "image" : "${aws_ecr_repository.training.repository_url}:latest",
+      "portMappings" : [
+        {
+          "name" : "gunicorn-port",
+          "containerPort" : 8000,
+          "hostPort" : 0,
+          "protocol" : "tcp",
+          "appProtocol" : "http"
+        }
+      ],
+      "essential" : true,
+      "environment" : [],
+      "mountPoints" : [],
+      "volumesFrom" : [],
+      "logConfiguration" : {
+        "logDriver" : "awslogs",
+        "options" : {
+          "awslogs-create-group" : "true",
+          "awslogs-region" : "us-east-1",
+          "awslogs-group" : aws_cloudwatch_log_group.training.name,
+          "awslogs-stream-prefix" : "ecs"
+        }
+      }
+    }
+  ])
+}
+
+# --- ECS Service ---
+resource "aws_ecs_service" "training" {
+  name            = "training"
+  cluster         = aws_ecs_cluster.main.id
+  task_definition = aws_ecs_task_definition.training.arn
+  desired_count   = 2
+
+  capacity_provider_strategy {
+    capacity_provider = aws_ecs_capacity_provider.training.name
+    base              = 1
+    weight            = 100
+  }
+
+  ordered_placement_strategy {
+    type  = "spread"
+    field = "attribute:ecs.availability-zone"
+  }
+
+  lifecycle {
+    ignore_changes = [desired_count]
+  }
+
+  # load_balancer {
+  #   target_group_arn = aws_lb_target_group.app.arn
+  #   container_name = "training"
+  #   container_port = 8000
+  # }
+
+  # depends_on = [aws_lb_target_group.app]
+}
+
```

Comment:
commented out code? do we need this?

Reply:
I believe I'll use service discovery instead of alb, but I'll change the code in the training issue

Reply:
ok. What's the difference between service discovery vs. alb? what are the tradeoffs? @andrewpeng02

Reply:
I believe they achieve similar purposes (routing to the containers in an ecs service), but service discovery is cheaper https://www.reddit.com/r/aws/comments/14ymt1w/help_me_understand_ecs_service_discovery/

Reply:
Very cool! Thanks for mentioning this. Could you flag this to the project leads to keep them in the loop :) Very interesting learning

Reply:
Like send a message in the general channel (or backend) channel and tag the project leads. This is a really interesting learning!